test/e2e/retry.go

```diff
@@ -33,17 +33,20 @@ const (
 	retryBackoffSteps           = 3
 )
 
-// retryWithTimeout retries the function until it returns true,
-// or a timeout is reached.
+// retryWithTimeout retries a function that returns an error until a timeout is reached
 func retryWithTimeout(interval, timeout time.Duration, fn func() error) error {
-	retryFn := func(fn func() error) func() (bool, error) {
-		return func() (bool, error) {
-			err := fn()
-			if err == nil {
-				return true, nil
-			}
-			return false, err
+	var pollError error
+	err := wait.PollImmediate(collectLogInterval, collectLogTimeout, func() (bool, error) {
+		pollError = nil
+		err := fn()
+		if err != nil {
+			pollError = err
+			return false, nil
```

Comment:
It looks like we're always returning `nil` for the error in this nested function. Would we ever want to return an error here?

Reply:
No, the purpose of this helper function is to absorb *every* error and only fail if the timeout has been reached.